src/Controller/Blockexplorer/BlockController.php

```diff
@@ -4,54 +4,68 @@
 
 use Adshares\AdsOperator\Controller\ApiController;
 use Adshares\AdsOperator\Document\Block;
+use Adshares\AdsOperator\Repository\BlockRepositoryInterface;
 use Swagger\Annotations as SWG;
 use Nelmio\ApiDocBundle\Annotation\Operation;
 use Nelmio\ApiDocBundle\Annotation\Model;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 
 class BlockController extends ApiController
 {
     /**
-     * @Operation(
-     *     summary="List of blocks",
-     *     tags={"Block"},
-     *
-     *      @SWG\Response(
-     *          response=200,
-     *          description="Returned when operation is successful",
-     *          @SWG\Schema(
-     *              type="array",
-     *              @SWG\Items(ref=@Model(type=Block::class))
-     *          )
-     *      )
-     * )
+     * BlockController constructor.
+     * @param BlockRepositoryInterface $repository
      */
-    public function listAction()
+    public function __construct(BlockRepositoryInterface $repository)
     {
-        return $this->response($this->serializer->serialize([], 'json'));
+        $this->repository = $repository;
     }
 
     /**
-     *
      * @Operation(
-     *     summary="Get block details",
-     *     tags={"Block"},
+     *     summary="List of blocks",
+     *     tags={"Blockexplorer"},
      *
-     *      @SWG\Response(
+     *     @SWG\Response(
      *          response=200,
      *          description="Returned when operation is successful",
-     *          @Model(type=Block::class)
+     *          @SWG\Schema(
+     *              type="array",
+     *              @SWG\Items(ref=@Model(type=Block::class))
+     *          )
+     *      ),
+     *     @SWG\Parameter(
+     *          name="sort",
+     *          in="query",
+     *          type="string",
+     *          description="The field used to order blocks"
      *      ),
-     *
      *      @SWG\Parameter(
-     *          name="id",
-     *          in="path",
+     *          name="order",
+     *          in="query",
      *          type="string",
-     *          description="Block identity"
+     *          description="The field used to sort blocks"
+     *      ),
+     *      @SWG\Parameter(
+     *          name="limit",
+     *          in="query",
+     *          type="int",
+     *          description="The field used to limit number of blocks"
+     *      ),
+     *      @SWG\Parameter(
+     *          name="offset",
+     *          in="query",
+     *          type="int",
+     *          description="The field used to specify blocks offset"
      *      )
      * )
+     *
+     * @param Request $request
+     * @return Response
      */
-    public function showAction()
+    public function listAction(Request $request): Response
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method "listAction" to simply inherit it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1185)